Dockerfile

```
@@ -38,4 +38,4 @@ RUN apt update -q \
     && echo $(curl -L https://dl.k8s.io/release/$KUBE_VERSION/bin/linux/$(dpkg --print-architecture)/kubectl.sha256) /usr/local/bin/kubectl | sha256sum --check \
     && chmod a+x /usr/local/bin/kubectl \
     && rm -rf /var/lib/apt/lists/* /var/log/* \
-    && find / -type d -name '*__pycache__' -prune -exec rm -rf {} \;
+    && find / -mount -type d -name '*__pycache__' -prune -exec rm -rf {} \;
```

It should be sufficient to only look into the workdir and prevent the issue your are having IMO
```suggestion
    && find . -type d -name '*__pycache__' -prune -exec rm -rf {} \;
```

---

Ah wait no... Actually there is already the no compile flag to pip so I guess we could simply remote this?

---

It's true that they shouldn't be created when pip is run with the  no-compile flag

But examining this in a  live docker container I realized that __pycache__ files are created anyway anyway.
They are created when the python3 python3-pip packages are installed. 
So the find -prune is still useful as it will free up 17mb.

I've updated my PR to only search the  /usr folder which is  where the __pycache__ files are located
Instead of searching the whole filesystem and ignoring mount points

---

Thanks for the additional insight! Could you squash your commit, apart from that lgtm.